garage/tf/algos/td3.py

```diff
@@ -0,0 +1,365 @@
+"""
+This module implements a TD3 model.
+
+TD3, or Twin Delayed Deep Deterministic Policy Gradient, uses actor-critic
+method to optimize the policy and reward prediction. Notably, it uses the
+minimum value of two critics instead of one to limit overestimation.
+"""
+from collections import deque
+
+import numpy as np
+import tensorflow as tf
+import tensorflow.contrib as tc
+
+import garage.misc.logger as logger
+from garage.misc.overrides import overrides
+from garage.tf.algos.off_policy_rl_algorithm import OffPolicyRLAlgorithm
+from garage.tf.misc import tensor_utils
+
+
+class TD3(OffPolicyRLAlgorithm):
+    """
+    Implementation of TD3.
+
+    Based on https://arxiv.org/pdf/1802.09477.pdf.
+
+    Example:
+        $ python garage/examples/tf/td3_pendulum.py
+    """
+
+    def __init__(self,
+                 env,
+                 qf2,
+                 replay_buffer,
+                 target_update_tau=0.01,
+                 policy_lr=1e-4,
+                 qf_lr=1e-3,
+                 policy_weight_decay=0,
+                 qf_weight_decay=0,
+                 policy_optimizer=tf.train.AdamOptimizer,
+                 qf_optimizer=tf.train.AdamOptimizer,
+                 clip_pos_returns=False,
+                 clip_return=np.inf,
+                 discount=0.99,
+                 max_action=None,
+                 name=None,
+                 **kwargs):
+        """
+        Construct class.
+
+        Args:
```

Comment:
I believe we are now moving constructor params to class docstring.

Reply:
Thanks for the heads up!

Reply:
please fix @naeioi